### PR TITLE
feat(seat): clarify seat states and action semantics

### PR DESCRIPTION
**Pull request by grepug**
> poker-types@1.0.0 build
> tsc

vite v7.3.1 building client environment for production...
transforming...
✓ 91 modules transformed.
rendering chunks...
computing gzip size...
dist/index.html                   0.93 kB │ gzip:   0.44 kB
dist/assets/index-B_0Z4jSd.css   68.14 kB │ gzip:  13.76 kB
dist/assets/index-DGbgwiZI.js   399.60 kB │ gzip: 123.95 kB
✓ built in 1.14s\n- 
> poker-server@0.0.1 test:e2e:playwright:smoke
> npm_config_proxy= npm_config_https_proxy= HTTP_PROXY= HTTPS_PROXY= ALL_PROXY= NO_PROXY='*' TEST_MODE=true ./node_modules/.bin/playwright test --project comprehensive-e2e --workers=1 --grep @critical --reporter=line

Running 6 tests using 1 worker

[1A[2K[1/6] [comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:2384:7 › Poker E2E - Test Suite 4: Edge Cases › @critical 4.3: Check When Bet Required - verify check button disabled when facing a bet

[1A[2K[comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:2384:7 › Poker E2E - Test Suite 4: Edge Cases › @critical 4.3: Check When Bet Required - verify check button disabled when facing a bet
ALICE: Connected to server http://localhost:3015

[1A[2KBOB: Connected to server http://localhost:3015

[1A[2KAlice creating room...

[1A[2KALICE: Room created: WCVWZU

[1A[2KALICE: Create room response: {success: true}

[1A[2KRoom created: WCVWZU

[1A[2KBob joining room...

[1A[2KBOB: Join room response: {success: true}

[1A[2KBoth players in room

[1A[2KAlice starting game...

[1A[2KBOB: Updating player on GAME_STARTED: {prev: 0, updated: 995}

[1A[2KALICE: Updating player on GAME_STARTED: {prev: 0, updated: 990}

[1A[2KBOB: Player turn: {playerId: 160a7cbc-fbfc-4920-a655-22115b0d03cd, playerName: Bob, timeLimit: 30000, currentBet: 10, minRaise: 10}

[1A[2KALICE: Player turn: {playerId: 160a7cbc-fbfc-4920-a655-22115b0d03cd, playerName: Bob, timeLimit: 30000, currentBet: 10, minRaise: 10}

[1A[2KALICE: Start game response: {success: true}

[1A[2KGame started

[1A[2KPre-flop: Bob waiting for turn...

[1A[2KPre-flop: Bob raising $50...

[1A[2KALICE: Player acted: {playerId: 160a7cbc-fbfc-4920-a655-22115b0d03cd, playerName: Bob, action: raise, amount: 50, displayKind: raise-to}

[1A[2KBOB: Player acted: {playerId: 160a7cbc-fbfc-4920-a655-22115b0d03cd, playerName: Bob, action: raise, amount: 50, displayKind: raise-to}

[1A[2KALICE: Player turn: {playerId: 0bda9479-b321-47b9-9720-49cf6416187c, playerName: Alice, timeLimit: 30000, currentBet: 60, minRaise: 50}

[1A[2KBOB: Player turn: {playerId: 0bda9479-b321-47b9-9720-49cf6416187c, playerName: Alice, timeLimit: 30000, currentBet: 60, minRaise: 50}

[1A[2KBOB: Action response: {success: true}

[1A[2KPre-flop: Alice facing Bob's raise...

[1A[2KAlice facing bet of $60

[1A[2K✓ Check button not present when Alice faces a bet

[1A[2K✓ Call button is enabled

[1A[2K✓ Fold button is enabled

[1A[2K✓ All-In button is enabled

[1A[2K
=== Test 4.3: Check validation verified - cannot check when facing a bet ===

[1A[2K[2/6] [comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:2572:7 › Poker E2E - Chip Conservation › @critical 6.1: Chip Conservation Throughout Hand - multiple hands in sequence

[1A[2K[comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:2572:7 › Poker E2E - Chip Conservation › @critical 6.1: Chip Conservation Throughout Hand - multiple hands in sequence
ALICE: Connected to server http://localhost:3015

[1A[2KBOB: Connected to server http://localhost:3015

[1A[2KAlice creating room...

[1A[2KALICE: Room created: 9NF9YB

[1A[2KALICE: Create room response: {success: true}

[1A[2KRoom created: 9NF9YB

[1A[2KBob joining room...

[1A[2KBOB: Join room response: {success: true}

[1A[2KBoth players in room

[1A[2K
=== Starting Hand ===

[1A[2KBOB: Updating player on GAME_STARTED: {prev: 0, updated: 995}

[1A[2KBOB: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 10, minRaise: 10}

[1A[2KALICE: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 10, minRaise: 10}

[1A[2KALICE: Start game response: {success: true}

[1A[2KALICE: Updating player on GAME_STARTED: {prev: 0, updated: 990}

[1A[2KGame started

[1A[2KBob calling...

[1A[2KBOB: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: call, displayKind: call-to, totalBetAfterAction: 10}

[1A[2KALICE: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: call, displayKind: call-to, totalBetAfterAction: 10}

[1A[2KALICE: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 10, minRaise: 10}

[1A[2KBOB: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 10, minRaise: 10}

[1A[2KBOB: Action response: {success: true}

[1A[2KAlice checking...

[1A[2KBOB: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 10}

[1A[2KBOB: Betting round complete, next round: FLOP

[1A[2KBOB: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 10}

[1A[2KALICE: Betting round complete, next round: FLOP

[1A[2KALICE: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Action response: {success: true}

[1A[2KFlop - Bob checking...

[1A[2KALICE: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KBOB: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KBOB: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KBOB: Action response: {success: true}

[1A[2KFlop - Alice checking...

[1A[2KBOB: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Betting round complete, next round: TURN

[1A[2KBOB: Betting round complete, next round: TURN

[1A[2KBOB: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Action response: {success: true}

[1A[2KTurn - Bob checking...

[1A[2KALICE: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KBOB: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KBOB: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KBOB: Action response: {success: true}

[1A[2KTurn - Alice checking...

[1A[2KBOB: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Betting round complete, next round: RIVER

[1A[2KBOB: Betting round complete, next round: RIVER

[1A[2KBOB: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Player turn: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KALICE: Action response: {success: true}

[1A[2KRiver - Bob checking...

[1A[2KALICE: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KBOB: Player acted: {playerId: 14ed39c3-75a6-486a-b963-90c96187ca33, playerName: Bob, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KBOB: Player turn: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, timeLimit: 30000, currentBet: 0, minRaise: 10}

[1A[2KBOB: Action response: {success: true}

[1A[2KRiver - Alice checking...

[1A[2KBOB: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KBOB: Betting round complete, next round: SHOWDOWN

[1A[2KBOB: Hand complete: {winners: Array(1), playerHands: Array(2), totalPot: 20, payouts: Array(1)}

[1A[2KALICE: Player acted: {playerId: a6c1841b-10f5-403d-a1d2-d7e09a72d1df, playerName: Alice, action: check, displayKind: check, totalBetAfterAction: 0}

[1A[2KALICE: Betting round complete, next round: SHOWDOWN

[1A[2KALICE: Hand complete: {winners: Array(1), playerHands: Array(2), totalPot: 20, payouts: Array(1)}

[1A[2KALICE: Action response: {success: true}

[1A[2KHand complete

[1A[2KFinal chips - Alice: 1010, Bob: 990

[1A[2K
=== Chip conservation verified throughout hand ===

[1A[2K[3/6] [comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:3315:7 › Poker E2E - Test Suite 5: Turn/Round Advancement › @critical 5.2: Round Progression - PRE_FLOP -> FLOP -> TURN -> RIVER -> SHOWDOWN

[1A[2K[4/6] [comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:4054:7 › Poker E2E - Test Suite 8: UI/UX Validation › @critical 8.5: Host Can Start Next Hand After Break

[1A[2K[5/6] [comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:4219:7 › Poker E2E - Test Suite 8: UI/UX Validation › @critical 8.8a: Seat Cards Stay Inside Table Bounds Across Viewports

[1A[2K[6/6] [comprehensive-e2e] › test/e2e/comprehensive-poker.spec.ts:4340:7 › Poker E2E - Test Suite 8: UI/UX Validation › @critical 8.12: Refresh Mid-Hand Automatically Reconnects Player Session

[1A[2K  6 passed (26.1s)

### CODE REVIEW

File: poker-server/src/events/events.gateway.ts [hunk 1]
```diff
@@ -748,15 +753,43 @@ export class EventsGateway implements OnGatewayConnection, OnGatewayDisconnect {
...
+          switch (data.action) {
+            case 'fold':
+              return 'fold' as PlayerActionDisplayKind;
+            case 'check':
+              return 'check' as PlayerActionDisplayKind;
+            case 'call':
+              return 'call-to' as PlayerActionDisplayKind;
+            case 'all-in':
+              return 'all-in-to' as PlayerActionDisplayKind;
+            case 'raise':
```
[the remark this answers is not in the document]
Comment (grepug) [reply poker-server/src/events/events.gateway.ts]: Good catch — fixed in 54861bb. We now derive emitted `action` and `displayKind` from the resolved post-action state (`updatedPlayer.lastAction`) rather than request intent, so coerced call/raise -> all-in paths emit all-in semantics correctly. Also set `amount` to `undefined` for resolved all-in to avoid stale requested raise amounts.
